### ISSUE TITLE
pip install fails: option --single-version-externally-managed not recognized

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#2769]
This issue was originally created at: 2011-06-01 00:31:15.
This issue was reported by: `akaihola`.
akaihola said at 2011-06-01 00:31:15
>On Ubuntu 11.04, this happens:
> 
>    $ sudo pip install scons
>    Downloading/unpacking scons
>      Downloading scons-2.1.0.alpha.20101125.tar.gz (570Kb): 570Kb downloaded
>      Running setup.py egg_info for package scons
>        
>    Installing collected packages: scons
>      Running setup.py install for scons
>        usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
>           or: -c --help [cmd1 cmd2 ...]
>           or: -c --help-commands
>           or: -c cmd --help
>        
>        error: option --single-version-externally-managed not recognized
>        Complete output from command /usr/bin/python -c "import 
>setuptools;__file__='/home/akaihola/build/scons/setup.py';exec(compile(open(__fi
>le__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-
>version-externally-managed --record /tmp/pip-6eO7cc-record/install-record.txt:
>        usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
>    
>       or: -c --help [cmd1 cmd2 ...]
>    
>       or: -c --help-commands
>    
>       or: -c cmd --help
>    
>    
>    
>    error: option --single-version-externally-managed not recognized
>    
>    ----------------------------------------
>    Command /usr/bin/python -c "import 
>setuptools;__file__='/home/akaihola/build/scons/setup.py';exec(compile(open(__fi
>le__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --single-
>version-externally-managed --record /tmp/pip-6eO7cc-record/install-record.txt 
>failed with error code 1
>    Storing complete log in /home/akaihola/.pip/pip.log

akaihola said at 2011-06-01 00:34:09
>The package was decompressed into $HOME/build, and by doing this I installed it 
>from there successfully:
> 
>    $ cd $HOME/build/scons
>    $ sudo python setup.py install

marcelhuberfoo said at 2013-06-18 04:27:03
>Try adding --egg to pip install options which solves the problem at least in my case.
>$ pip install --egg scons
>Downloading/unpacking scons
>  Downloading scons-2.3.0.tar.gz (599kB): 599kB downloaded
>  Running setup.py egg_info for package scons
>    
>Installing collected packages: scons
>  Running setup.py install for scons
>    changing mode of build/scripts/scons from 644 to 755
>    changing mode of build/scripts/sconsign from 644 to 755
>    changing mode of build/scripts/scons-time from 644 to 755
>    changing mode of build/scripts/scons.bat from 644 to 755
>    
>    Installed SCons library modules into /home/blub/.virtualenvs/EnvPython2Scons/lib/scons-2.3.0
>    Installed SCons scripts into /home/blub/.virtualenvs/EnvPython2Scons/bin
>    Installed SCons man pages into /home/blub/.virtualenvs/EnvPython2Scons/man/man1
>Successfully installed scons
>Cleaning up...
> 
>~ $ which scons
>/home/blub/.virtualenvs/EnvPython2Scons/bin/scons
> 
>~ $ scons --version
>SCons by Steven Knight et al.:
>	script: v2.3.0, 2013/03/03 09:48:35, by garyo on reepicheep
>	engine: v2.3.0, 2013/03/03 09:48:35, by garyo on reepicheep
>	engine path: ['/home/blub/.virtualenvs/EnvPython2Scons/lib/scons-2.3.0/SCons']
>Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013 The SCons Foundation

bdbaddog said at 2014-03-02 16:41:47
>Should be fixed by this pull request:
>https://bitbucket.org/scons/scons/pull-request/113/fix-for-bug-2769-which-should-allow-scons/diff
> 
>Once this is merged and the next version is released 2.3.2 most likely, then this should be resolved

bdbaddog said at 2014-03-02 20:25:55
>Take ownership

jonathanrrogers said at 2014-03-17 14:20:20
>Adding --egg to the pip call does work around this problem for me, but the pull
>request did not.

bdbaddog said at 2014-03-25 10:03:15
>*** Issue 2935 has been marked as a duplicate of this issue. ***

Votes for this issue: 10.

**Comment by Hecatron** [comment]
I recently ran into this issue
the pull request / commit works, but it looks as if the pull request was never accepted

  * https://bitbucket.org/scons/scons/pull-requests/113/fix-for-bug-2769-which-should-allow-scons/diff
  * https://bitbucket.org/bdbaddog/scons/commits/c89c850e906fa0e2cc2bd70c5bf4dabfe4241e8d?at=default

**Comment by tueda** [comment]
I ran into this issue when I put `install_requires=['SCons']` in a `setup.py` of my package (to build binaries internally executed from Python files).

**Comment by bdbaddog** [comment]
This should be resolved as we're producing wheels for pypi for the last few releases and going forward. Please reopen if it's not resolved.
